src/lib/get-helia.ts

```diff
@@ -0,0 +1,122 @@
+import { noise } from '@chainsafe/libp2p-noise'
+import { yamux } from '@chainsafe/libp2p-yamux'
+import { trustlessGateway, bitswap } from '@helia/block-brokers'
+import { createDelegatedRoutingV1HttpApiClient } from '@helia/delegated-routing-v1-http-api-client'
+import { createHeliaHTTP } from '@helia/http'
+import { type BlockBroker } from '@helia/interface'
+import { httpGatewayRouting, delegatedHTTPRouting, libp2pRouting } from '@helia/routers'
+import { createVerifiedFetch, type VerifiedFetch } from '@helia/verified-fetch'
+import { generateKeyPair } from '@libp2p/crypto/keys'
+import { dcutr } from '@libp2p/dcutr'
+import { identify, identifyPush } from '@libp2p/identify'
+import { keychain } from '@libp2p/keychain'
+import { ping } from '@libp2p/ping'
+import { webSockets } from '@libp2p/websockets'
+import { webTransport } from '@libp2p/webtransport'
+import { dns } from '@multiformats/dns'
+import { dnsJsonOverHttps } from '@multiformats/dns/resolvers'
+import { createHelia, type Helia, type Routing } from 'helia'
+import { createLibp2p, type Libp2pOptions } from 'libp2p'
+import * as libp2pInfo from 'libp2p/version'
+import { contentTypeParser } from './content-type-parser.js'
+import type { ConfigDb } from './config-db'
+import type { ComponentLogger } from '@libp2p/logger'
+
+export async function getVerifiedFetch (config: ConfigDb, logger: ComponentLogger): Promise<VerifiedFetch> {
+  const log = logger.forComponent('get-verified-fetch')
+  log(`config-debug: got config for sw location ${self.location.origin}`, config)
+
+  if (!config.enableRecursiveGateways && !config.enableGatewayProviders && !config.enableWss && !config.enableWebTransport) {
+    // crude validation
+    throw new Error('Config is invalid. At least one of the following must be enabled: recursive gateways, gateway providers, wss, or webtransport.')
+  }
```

Comment:
Can we do this validation when config save button is enabled/clicked instead of here in `getVerifiedFetch`? I believe we have logic for that already

Reply:
Yeah good point. That was a TODO.We don't have global validation implemented. But I'll take care of that.

Reply:
I thought we had validation before save was clicked.. the button was disabled if inputs were invalid.. we can save until later if that's the case.